aea/decision_maker/base.py

```diff
@@ -841,3 +847,35 @@ def _handle_state_update_message(
                 quantities_by_good_id=state_update_message.quantities_by_good_id,
             )
             self._ownership_state = new_ownership_state
+
+    def _handle_tx_message_for_deployment(self, tx_message) -> None:
```

Comment:
We should discuss this in person. I think it could be merged with the signing handler

Reply:
I think the skill should send a `TransactionMessage` with performative `PROPOSE_FOR_SIGNING` to the DM which contains the tx. The actual deployment should then be done from the skill once it receives the callback.